### DISCUSSION TITLE
Default update behavior for unstable -> stable versions

### DISCUSSION THREAD

**Discussion by stekern** [renovatebot/renovate#13849] [Request Help] [answered]
What rules do Renovate use by default when it comes to updating a given package from an unstable to a stable version?

If I have a package installed with version `1.0.0-alpha.1`, and a version `1.0.1` or `1.1.0` is released, will Renovate by default try to update this? If so, is there an easy way to disable this?

I see there are some open issues (https://github.com/renovatebot/renovate/issues/4723, https://github.com/renovatebot/renovate/issues/4852) related to these scenarios, but it's not entirely clear to me what the current behavior is.

**Comment by rarkins** [comment, marked as answer]
Renovate will always try to update from unstable to higher stable. To prevent this you'd need to either ignore that dep completely (e.g. `ignoreDeps`), or add a packageRule with `matchPackageNames`, `matchCurrentVersion` (optional)`, and `allowedVersions` (e.g. `^1.0.0-alpha.1` or `<1.0.0`).

**Reply by stekern** [reply]
Thanks for the quick reply and clarification.

Would the following rule then enforce that an alpha version can only be updated to another alpha version?
```json
{
   "matchPackageNames": ["my-package"],
   "matchCurrentVersion": "/^.*-alpha/",
   "allowedVersions": "/^.*-alpha/"
}
```

**Reply by rarkins** [reply]
I think so
